File: problems/70/Solution.swift
class Solution {
    func climbStairs(_ n: Int) -> Int {
        var steps = [1, 1]
        if n >= 2 {
            for i in 2...n {
                steps.append(steps[i-1] + steps[i-2])
            }
        }
        return steps[n]
    }
}

File: problems/70/Solution.java
package tw.idv.rchu.leetcode;

public class Solution {
    public int climbStairs(int n) {
        int[] steps = new int[n+2];

        steps[0] = 1;
        steps[1] = 1;
        for (int i = 2; i <= n; i++) {
            steps[i] = steps[i-1] + steps[i-2];
        }

        return steps[n];
    }
}